detect_secrets/filters/heuristic.py

```diff
@@ -181,7 +181,7 @@ def _get_indirect_reference_regex() -> Pattern:
     #       [^\v]*      ->  Something except line breaks
     #       [\]\)]      ->  End of indirect reference: ] or )
     #   )
-    return re.compile(r'([^\v=!:]*)\s*(:=?|[!=]{1,3})\s*([\w.-]+[\[\(][^\v]*[\]\)])')
+    return re.compile(r'(\w+)\s*(:=?|[!=]{1,3})\s*([\w.-]+[\[\(][^\v]*[\]\)])')
```

Comment:
nice catch  :)

comment on line 174 needs updating

do you know if there are unit tests covering this regex? I'd be curious what practical effect this change has (lines that it matches now that it didn't before or vice versa)

Reply:
also given how this regex is used, I feel like option 1 is an attractive idea, I can't imagine why you'd want this heuristic to apply to lines that long, and since it's a heuristic having false negatives doesn't matter too much

Reply:
I'm also trying this on regex101.com and it seems to still have just as much backtracking as it did before

Reply:
There are unit tests but they do not have much coverage - only 3 cases:
`    'line, result',
    (
        ('secret = get_secret_key()', True),
        ('secret = request.headers["apikey"]', True),
        ('secret = hunter2', False),
    ),
)`

Reply:
I'm wondering if we apply both approaches. Regex update and a cap on line length - maybe 20k characters?

Reply:
It sounds like the kinds of lines this is trying to catch should be pretty short (similar to `secret = get_secret_key()`, function calls and the like), so I'd put an even lower limit, 1K or so (and in practice I wouldn't expect to see anything much above 100 chars)

Reply:
Updated